packages/protocol/contracts/team/airdrop/MerkleClaimable.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { OwnableUpgradeable } from
+    "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import { MerkleProofUpgradeable } from
+    "lib/openzeppelin-contracts-upgradeable/contracts/utils/cryptography/MerkleProofUpgradeable.sol";
+
+import { Proxied } from "../../common/Proxied.sol";
+
+/// @title MerkleClaimable
+/// Contract for managing Taiko token airdrop for eligible users
+// TODO(dani): add claimStart and claimEnd timestamp so claim() can only be
+// called between these two timestamps.
+// Better to add non-reentrance guard to claim().
```

Comment:
`Line 33` already prevents reentrancy, because the only time a contract can be invoked is from the call _claimWithData() but by then (in case of a reentrancy) the hash is already registered as claimed.